src/lib/components/Swap/SwapButton.tsx

```diff
@@ -184,7 +184,7 @@ export default function SwapButton({ disabled }: SwapButtonProps) {
   }, [addTransaction, inputTradeCurrencyAmount, outputTradeCurrencyAmount, setDisplayTxHash, swapCallback, tradeType])
 
   const ButtonText = useCallback(() => {
-    if (wrapError !== WrapError.NO_ERROR) {
+    if (wrapError !== WrapError.NO_ERROR && (wrapType === WrapType.WRAP || wrapType === WrapType.UNWRAP)) {
```

Comment:
nit: would prefer these to be reversed, as it makes sense to only check for an error if the wrapType is applicable.

Reply:
Yep